--- ash-linux-0.2/mkinit.c.orig	Tue Aug 24 06:12:51 1993
+++ ash-linux-0.2/mkinit.c	Thu Sep 23 16:23:46 1999
@@ -349,7 +350,7 @@
 		if (! amiddecls)
 			addchar('\n', &decls);
 		q = NULL;
-		for (p = line1 + 6 ; *p != '=' && *p != '/' ; p++);
+		for (p = line1 + 6 ; *p != '\n' && *p != '=' && *p != '/' ; p++);
 		if (*p == '=') {		/* eliminate initialization */
 			for (q = p ; *q && *q != ';' ; q++);
 			if (*q == '\0')
